packaging/installer/netdata-updater.sh

```diff
@@ -372,16 +372,27 @@ get_netdata_latest_tag() {
 newer_commit_date() {
   info "Checking if a newer version of the updater script is available."
 
+  commit_check_url="https://api.github.com/repos/netdata/netdata/commits?path=packaging%2Finstaller%2Fnetdata-updater.sh&page=1&per_page=1"
+  python_version_check="from __future__ import print_function;import sys,json;data = json.load(sys.stdin);print(data[0]['commit']['committer']['date'] if isinstance(data, list) else '')"
+
   if command -v jq > /dev/null 2>&1; then
-    commit_date="$(_safe_download "https://api.github.com/repos/netdata/netdata/commits?path=packaging%2Finstaller%2Fnetdata-updater.sh&page=1&per_page=1" /dev/stdout | jq '.[0].commit.committer.date' | tr -d '"')"
+    commit_date="$(_safe_download "${commit_check_url}" /dev/stdout | jq '.[0].commit.committer.date' 2>/dev/null | tr -d '"')"
   elif command -v python > /dev/null 2>&1;then
-    commit_date="$(_safe_download "https://api.github.com/repos/netdata/netdata/commits?path=packaging%2Finstaller%2Fnetdata-updater.sh&page=1&per_page=1" /dev/stdout | python -c 'from __future__ import print_function;import sys,json;print(json.load(sys.stdin)[0]["commit"]["committer"]["date"])')"
+    commit_date="$(_safe_download "${commit_check_url}" /dev/stdout | python -c "${python_version_check}")"
   elif command -v python3 > /dev/null 2>&1;then
-    commit_date="$(_safe_download "https://api.github.com/repos/netdata/netdata/commits?path=packaging%2Finstaller%2Fnetdata-updater.sh&page=1&per_page=1" /dev/stdout | python3 -c 'from __future__ import print_function;import sys,json;print(json.load(sys.stdin)[0]["commit"]["committer"]["date"])')"
+    commit_date="$(_safe_download "${commit_check_url}" /dev/stdout | python3 -c "${python_version_check}")"
   fi
 
   if [ -z "${commit_date}" ] ; then
-    commit_date="9999-12-31T23:59:59Z"
+    commit_date="9999999999"
+  elif [ "$(uname)" = "Linux" ]; then
+    commit_date="$(date -d "${commit_date}" +%s)"
+  else # assume BSD-style `date` if we are not on Linux
+    commit_date="$(date -j -f "%Y-%m-%dT%H:%M:%SZ" "${commit_date}" +%s 2>/dev/null)"
```

Comment:
Doesn't work on macOS with installed [coreutils](https://formulae.brew.sh/formula/coreutils)

```cmd
~ $ which date
/opt/homebrew/opt/coreutils/libexec/gnubin/date
~ $ echo $(date -j -f "%Y-%m-%dT%H:%M:%SZ" "${commit_date}" +%s)
date: invalid option -- 'j'
Try 'date --help' for more information.

~ $ echo $(/bin/date -j -f "%Y-%m-%dT%H:%M:%SZ" "${commit_date}" +%s)
253402293599
```

Reply:
Of course it doesn’t...

Because it would apparently be too sensible to assume that Homebrew sanely behaves like BSD here and installs with the name `gdate`...

Reply:
```cmd
~ $ ls -l /opt/homebrew/opt/coreutils/libexec/gnubin/date
lrwxr-xr-x 15 ilyam 24 Sep  2021 /opt/homebrew/opt/coreutils/libexec/gnubin/date -> ../../bin/gdate
```